CMakeLists.txt

```diff
@@ -0,0 +1,52 @@
+cmake_minimum_required(VERSION 3.13)
+include(src/cmake/projectVersionDetails.cmake)
+project(log4cxx VERSION ${log4cxx_VER} LANGUAGES CXX)
+include(CTest)
+
+# FindAPR and FindAPR-util are not provided by APR and APR-Util so source them locally
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/src/cmake")
+
+# Find Apache Runtime
+find_package(APR QUIET)
+# If APR find module sets the cache, the following will do nothing
+find_path(APR_INCLUDE_DIR apr.h PATH_SUFFIXES apr-1)
+find_library(APR_LIBRARIES NAMES libapr-1 apr-1)
+
+# Find Apache Runtime Utilities
+find_package(APR-Util QUIET)
+# If APR-UTIL find module sets the cache, the following will do nothing
+find_path(APR_UTIL_INCLUDE_DIR apu.h PATH_SUFFIXES apr-1)
+find_library(APR_UTIL_LIBRARIES NAMES libaprutil-1 aprutil-1)
```

Comment:
Shouldn't this use FindPkgConfig and pkg_check_modules on Unix systems instead of find_package?

Reply:
DO APR and APR-UTIL provide support pkg-config?

Reply:
As far as I'm aware they do; I just tried this on Debian 10(buster), and it works properly though.  I'm not sure what is best.  Maybe this is better as a fallback if pkg_check_modules fails?

Reply:
The best practices from Professional CMake book include the statement:
"Use of the FindPkgConfig module should be reserved only for those situations where find_package()
is not suitable."

I am inclined to stick with find_package if that is OK.